src/main/java/io/jenkins/plugins/kubernetes/NoDelayProvisionerStrategy.java

```diff
@@ -0,0 +1,72 @@
+package io.jenkins.plugins.kubernetes;
+import hudson.Extension;
+import hudson.model.Label;
+import hudson.model.LoadStatistics;
+import hudson.slaves.Cloud;
+import hudson.slaves.NodeProvisioner;
+import jenkins.model.Jenkins;
+import org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud;
+
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Implementation of {@link NodeProvisioner.Strategy} which will provision a new node immediately as
+ * a task enter the queue.
+ * In kubernetes, we don't really need to wait before provisioning a new node,
+ * because kubernetes agents can be started and destroyed quickly
+ *
+ * @author <a href="mailto:root@junwuhui.cn">runzexia</a>
+ */
+@Extension(ordinal = 100)
+public class NoDelayProvisionerStrategy extends NodeProvisioner.Strategy {
+
+    private static final Logger LOGGER = Logger.getLogger(NoDelayProvisionerStrategy.class.getName());
+    private static final boolean DISABLE_NODELAY_PROVISING = Boolean.valueOf(
+            System.getProperty("io.jenkins.plugins.kubernetes.disableNoDelayProvisioning"));
+
+    @Override
+    public NodeProvisioner.StrategyDecision apply(NodeProvisioner.StrategyState strategyState) {
+        if (DISABLE_NODELAY_PROVISING) {
+            LOGGER.log(Level.FINE, "Provisioning not complete, NoDelayProvisionerStrategy is disabled");
+            return NodeProvisioner.StrategyDecision.CONSULT_REMAINING_STRATEGIES;
+        }
+
+        final Label label = strategyState.getLabel();
+
+        LoadStatistics.LoadStatisticsSnapshot snapshot = strategyState.getSnapshot();
+        int availableCapacity =
+                snapshot.getAvailableExecutors()   // live executors
+                        + snapshot.getConnectingExecutors()  // executors present but not yet connected
+                        + strategyState.getPlannedCapacitySnapshot()     // capacity added by previous strategies from previous rounds
+                        + strategyState.getAdditionalPlannedCapacity();  // capacity added by previous strategies _this round_
+        int currentDemand = snapshot.getQueueLength();
+        LOGGER.log(Level.FINE, "Available capacity={0}, currentDemand={1}",
+                new Object[]{availableCapacity, currentDemand});
+        if (availableCapacity < currentDemand) {
```

Comment:
Add this condition inside the loop as well, this will prevent unneeded calls to `Cloud#provision`

Reply:
https://github.com/jenkinsci/kubernetes-plugin/pull/598/files#diff-d96aedf6c35e2abaa077d40cd6991434R60
Would break the loop here, I think there is no need for additional judgment conditions?

Reply:
you're right, missed it.